controllers/storagecluster/storageclasses.go

```diff
@@ -139,7 +139,7 @@ func (r *StorageClusterReconciler) createStorageClasses(sccs []StorageClassConfi
 				skippedSC = append(skippedSC, sc.Name)
 				continue
 			}
-		case strings.Contains(sc.Name, "-cephfs") && !scc.isClusterExternal:
+		case (strings.Contains(sc.Name, "-cephfs") || strings.Contains(sc.Provisioner, "cephfs.csi.ceph.com")) && !scc.isClusterExternal:
```

Comment:
instead of `cephfs.csi.ceph.com` check for `<storagecluster-namespace>.cephfs.csi-ceph.com`

Reply:
Ok, for that I think we can use the `clusterID` from the SC for getting the namespace as it contains only the namespace of the storagecluster. Else, we will have to fetch it explicitly, wdyt @iamniting ?

Reply:
clusterID cannot be used even though it looks like a namespace name, but it can be any arbitrary name. My suggestion is to use a namespace where the Rook operator is running.

Reply:
@Madhu-1 What problem do you see if we just use `cephfs.csi.ceph.com`?

Reply:
Oh ok, you mean do something like this, right?
```
podList := &corev1.PodList{}
	err := r.Client.List(context.TODO(), podList, client.MatchingLabels{"app": "rook-ceph-operator"})
	if err != nil {
		r.Log.Error(err, "Failed to list rook-ceph-operator pod")
		return err
	}
	// Get the namespace in which the rook operator is running
	scNS := podList.Items[0].Namespace
```

Reply:
That can be used based on the assumption that no one is running any csi driver with that prefix for there CSI drivers.

Reply:
You can use the namespace of the storagecluster in your case or the ocs-operator namespace can also be used.

Reply:
We can get the namespace easily See this https://github.com/red-hat-storage/ocs-operator/blob/main/main.go#L121-L125

Reply:
Made the update to use the ocs-operator namespace

Reply:
Seems like the `OPERATOR_NAMESPACE` env variable is not set in the CI 🤔

Reply:
I think in that case we can add a double check for ns fetch, i.e, if the env variable is not there we can fetch the namespace from the `ocs-operator` namespace.

Reply:
Updated the test to set the `OPERATOR_NAMESPACE` env, looks good now.